File: javaproject/app/src/main/java/com/marcosvbras/robomarket/profile/viewmodel/EditProfileViewModelFactory.java
package com.marcosvbras.robomarket.profile.viewmodel;

import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;
import android.support.annotation.NonNull;

import com.marcosvbras.robomarket.interfaces.BaseActivityCallback;

import javax.inject.Inject;

public class EditProfileViewModelFactory implements ViewModelProvider.Factory {

    private BaseActivityCallback activityCallback;

    @Inject
    public EditProfileViewModelFactory(@NonNull BaseActivityCallback activityCallback) {
        this.activityCallback = activityCallback;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new EditProfileViewModel(activityCallback);
    }
}

File: javaproject/app/src/main/java/com/marcosvbras/robomarket/flows/profile/viewmodel/EditProfileViewModelFactory.kt
package com.marcosvbras.robomarket.flows.profile.viewmodel

import android.arch.lifecycle.ViewModel
import android.arch.lifecycle.ViewModelProvider
import com.marcosvbras.robomarket.interfaces.ActivityCallback

@Suppress("UNCHECKED_CAST")
class EditProfileViewModelFactory(private val activityCallback: ActivityCallback) : ViewModelProvider.NewInstanceFactory() {

    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return EditProfileViewModel(activityCallback) as T
    }

}